src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -364,6 +360,8 @@ public static void ConfigureConnectionStringManifestPublisher(ApplicationModel.I
 
         public static ApplicationModel.ParameterResource CreateGeneratedParameter(IDistributedApplicationBuilder builder, string name, bool secret, ApplicationModel.GenerateParameterDefault parameterDefault) { throw null; }
 
+        public static ApplicationModel.ParameterResource CreateParameter(IDistributedApplicationBuilder builder, string name, bool secret) { throw null; }
```

Comment:
I think this API will be confusing. From an intellisense point of view when you type `builder.` you are going to see `AddParameter` and `CreateParameter`. What is the difference?

Reply:
One doesn't get added to the app model.

Reply:
> From an intellisense point of view when you type builder. you are going to see AddParameter and CreateParameter. 

This isn't an extension method. It is static, so you'd need to invoke it with `ParameterResourceBuilderExtensions.`.

This follows the existing `CreateXXXParameter` methods in this class.

Reply:
These methods should really be on Parameter.

Reply:
@davidfowl - do you think we should fix that now? Or just continue like this for 9.3, and someday maybe obsolete these and make static methods on ParameterResource?

Reply:
[Consider making CreateParameter methods as static on ParameterResource (dotnet/aspire#9362)](https://github.com/dotnet/aspire/issues/9362)